lib/Dialect/TritonNvidiaGPU/Transforms/PlanCTA.cpp

```diff
@@ -240,8 +240,20 @@ bool CTAPlanner::processDot(triton::FuncOp &funcOp) {
   // TODO: This is a naive implementation and should be refactored
   auto getCTATiling = [](int64_t M, int64_t N, int64_t K,
                          unsigned numCTAs) -> std::pair<unsigned, unsigned> {
-    unsigned splitM = std::clamp<unsigned>(M / 64, 1, numCTAs);
-    unsigned splitN = numCTAs / splitM;
+    // perfer a larger chunk size, at most 128; first assign splitM.
+    unsigned chunk_m = 128;
+    auto isLegal = [](unsigned chunk){
+      return chunk >= 64;
+    };
+    assert(isLegal(N) && "numCTAs > 1 now only supports cases with BLOCK_N >= 64");
+    unsigned splitM, splitN;
+    for(;isLegal(chunk_m); chunk_m /= 2) {
+      splitM = std::clamp<unsigned>(M / chunk_m, 1, numCTAs);
+      splitN = numCTAs / splitM;
+      if (isLegal(N / splitN)) // chunk_n;
+        break;
+    }
+    assert(isLegal(N / splitN) && isLegal(chunk_m) && "Too large numCTAs, try to reduce it.");
```

Comment:
Why do we need to check `N / splitN >= 64` ? Is it from the constraint of GMMA instruction shape? But the constraint is only in M dim, I think?

Does `N / splitN >= 64` means `(N / splitN) % 64 == 0` ?

Thanks for the explanation in advance.

Reply:
After offline discussion, have removed these unnecessary asserts. Thanks!